tests/test_split_command.rs

```diff
@@ -132,3 +132,131 @@ fn test_split_by_paths() {
     A file2
     "###);
 }
+
+#[test]
+fn test_split_empty_child() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file1"), "foo").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "original parent"]);
+
+    std::fs::write(repo_path.join("file2"), "foo").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "target commit"]);
+
+    std::fs::write(repo_path.join("file3"), "foo").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["describe", "-m", "first original child"]);
+
+    test_env.jj_cmd_success(
+        &repo_path,
+        &["new", "-r", "@-", "-m", "second original child"],
+    );
+    std::fs::write(repo_path.join("file3"), "foo").unwrap();
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &[
+            "log",
+            "-T",
+            r#"change_id.short() if(empty, " (empty) ", " ") description"#,
+        ],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    @ d043564ef936 second original child
+    | o ffdaa62087a2 first original child
+    |/  
+    o 8e4fac809cbb target commit
```

Comment:
Optional: You could make "target commit" have more than one parent.

Reply:
Good idea. Done.